### ISSUE TITLE
Tips for extending wireless with external access point...?

### ISSUE THREAD

**Issue by rustyb** [AmericanRedCross/posm#267]
Hi Folks - Have been trying unsuccessfully to extend the wireless of the POSM using a TP-Link with dd-wrt installed for the last few days. Anyone have any tips or documentation for setting this up?

TP-Link connected via hotplug on the back of the POSM. The gateway on the TP-Link is set to `172.16.2.1`. 

I can ssh into the device and connect through to posm.local. However no joy with being able to posm.io.

Standard NUC with the same setup as per home POSM readme. No custom wireless card. POSM set to bridge mode.

Any advice on how to get this up an running?

**Comment by dalekunce** [comment]
My guess would be that you need to create a host record for posm.io on the TP-Link. The upgraded wireless card for POSM/NUC is also very good and can handle not only a lot more clients but has a little better range as well in our test.

**Comment by mojodna** [comment]
tl;dr: I think it's possible in a couple of different ways, but will require re-configuration, some of which will need to be done by hand.

## Method 1

`posm.io` doesn't resolve correctly because the TP-Link isn't using the POSM for upstream DNS resolution. You may be able to get it working by configuring the TP-Link to use POSM's IP address as its DNS server (I _think_ it binds DNS services to the ethernet / WAN interface).

There are a few services that won't work in this mode (Samba file sharing for one, as it only binds to LAN interfaces), as users will be connecting to POSM via its WAN (internet) port (the wifi acts as a LAN port).

Captive portal-like behavior is possible in this mode (by putting https://github.com/AmericanRedCross/posm-build/blob/master/kickstart/etc/dnsmasq-captive.conf and https://github.com/AmericanRedCross/posm-build/blob/master/kickstart/etc/nginx-captive.conf in place and bypassing the rest of `captive-deploy.sh`) where clients will automatically be redirected to POSM.

It can be bypassed by setting alternate DNS servers on clients (e.g. 8.8.8.8), since traffic isn't actually passing through POSM.

## Method 2

When doing initial bootstrapping of the system, LAN + WAN ports can be configured: https://github.com/AmericanRedCross/posm-build/blob/master/kickstart/etc/settings#L11-L12

This was originally added for devices with multiple ethernet ports, but re-configuring the single ethernet port to be used as a LAN port instead will allow "wired clients" (including those from the TP-Link) to connect on the POSM's "internal" network.

Switching to this configuration has one significant gotcha/opportunity: POSM provides DHCP (and DNS) services for clients connecting on its LAN port(s), so connecting a designated LAN port to the TP-Link will trigger a situation where there will be 2 DHCP servers active. To work around this (and to enable captive portal mode), re-configure the TP-Link as an access point only (no routing/DNS/DHCP), obtaining an IP address (and DNS settings) from the POSM (via its wired LAN port).

Captive portal mode will work in this mode.
